pkg/cloud/aws/instances.go

```diff
@@ -0,0 +1,52 @@
+package aws
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	azDown "github.com/litmuschaos/litmus-go/pkg/kube-aws/az-down/types"
+	"github.com/litmuschaos/litmus-go/pkg/log"
+	"strings"
+)
+
+// returns all instances by name, id and az for region that cluster identifier
+func GetAllInstancesInRegion(ec2Svc *ec2.EC2) ([]*azDown.InstanceDetails, error){
+
+	instanceData, err := ec2Svc.DescribeInstances(nil)
+
+	if err != nil {
+		log.Error(fmt.Sprintf("Could not list instances in region %s", err.Error()))
+	}
+
+	reservations := instanceData.Reservations
+
+	instances := make([]*azDown.InstanceDetails, len(reservations))
+	for i, res := range reservations {
+		instance := azDown.InstanceDetails{
+			Name: *res.Instances[0].InstanceId,
```

Comment:
this could be a array index out of range problem. For loop, loops for reservations that doesn’t guarantee that instances need to be there. Separate check/loop for instances within each reservation might be a better choice.